netconf/session.go

```diff
@@ -44,7 +48,23 @@ func (s *Session) Exec(methods ...RPCMethod) (*RPCReply, error) {
 		return nil, err
 	}
 
-	rawXML, err := s.Transport.Receive()
+	var rawXML []byte
```

Comment:
I'd rather see context.WithTimeout used here which can cancel the process or socket (depending on what transport is being used) that this.

https://golang.org/pkg/context/#WithTimeout